src/test/java/com/onaio/steps/adapter/HouseholdAdapterTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016. World Health Organization
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.onaio.steps.adapter;
+
+import android.view.View;
+import android.widget.TextView;
+
+import com.onaio.steps.R;
+import com.onaio.steps.activities.HouseholdListActivity;
+import com.onaio.steps.adapters.HouseholdAdapter;
+import com.onaio.steps.model.Household;
+import com.onaio.steps.model.InterviewStatus;
+
+import junit.framework.Assert;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Config(emulateSdk = 16, manifest = "src/main/AndroidManifest.xml")
+@RunWith(RobolectricTestRunner.class)
+public class HouseholdAdapterTest {
+
+    private HouseholdAdapter adapter;
+    private List<Household> householdList;
+
+    @Before
+    public void setUp() {
+        householdList = new ArrayList<>();
+        householdList.add(getHousehold());
+
+        HouseholdListActivity householdListActivity = Mockito.spy(Robolectric.buildActivity(HouseholdListActivity.class).create().resume().get());
+        adapter = new HouseholdAdapter(householdListActivity, householdList);
+    }
+
+    @Test
+    public void testGetCountShouldReturnOne() {
+        Assert.assertEquals(1, householdList.size());
+    }
+
+    @Test
+    public void testGetItemShouldVerifyTheData() {
+        Household household = (Household) adapter.getItem(0);
+        verify(household);
+    }
+
+    @Test
+    public void testGetItemIdShouldReturnOneAsLong() {
+        Assert.assertEquals(1, adapter.getItemId(0));
+    }
+
+    @Test
+    public void testGetViewShouldVerifyViewData() {
+        View itemView = adapter.getView(0, null, null);
+
+        TextView householdName = (TextView) itemView.findViewById(R.id.main_text);
+
+        Assert.assertEquals("HHID:" + NAME, householdName.getText().toString());
+    }
+
+    private void verify(Household household) {
```

Comment:
If we are not reusing this method, we can move the code to the specific test for better readability

Reply:
done method removed.